### ISSUE TITLE
Feature Request: Documenting REPLY

### ISSUE THREAD

**Issue by hyperupcall** [reconquest/shdoc#38]
I think it would be useful to be able to document what return variables like `REPLY` and `REPLY1`, `REPLY1`, etc. mean.

It could possibly look like the following

```sh
# @reply REPLY string The full path to your root project folder
```

Thoughts?

**Comment by hyperupcall** [comment]
Oops, probably should have been more specific - I mean setting global variables to be used as "return" variables

For example, the `read` builtin sets whatever it reads from stdin to a variable called `REPLY` if a variable name isn't explicitly specified as an argument. Most of [my functions](https://github.com/hyperupcall/basalt/blob/d4ca893c6b3a3f25874726fba0036e2cfd1f6566/pkg/lib/util/util.sh#L286) have this pattern since its faster than 
`"$(fn)"` and it's easier to "return" multiple things

**Comment by kovetskiy** [comment]
Don't really get the point of this REPLY variable, why you don't just use any other name? Like `BLAH` since you can modify it from the inside of bash function and use this BLAH after that?

I'm sorry if I don't get something from the bash documentation

**Comment by hyperupcall** [comment]
You're right that it's not really special and other names could be used. I only used `REPLY` since both `read` and `select` both use it, and it seemed pretty semantic 

So for your function that sets `BLAH` so it can be used outside of the function, it would be nice to be able to document that:

```sh
# @description Lorem ipsum
# @reply BLAH string Meaning of blah
get_something_cool() {
    ...
    BLAH="somethingmeaningful"
}
```

Personally, I don't set global variables other than `REPLY*` in Bash functions, since with my coding style, I set it to a local variable right after anyways, and the meaning of `REPLY` is usually implied by the function name

```sh
main() {
    get_something_cool
    local something_cool="$REPLY"
}
```

**Comment by hyperupcall** [comment]
Of course if you wish, we can not use `@reply` and go with something like `@sets` instead. Actually, I like the sound of `@sets` a lot better, since it represents what we're actually doing: setting global variables
